src/StateSmith/Output/Sim/SimWebGenerator.cs

```diff
@@ -144,21 +144,51 @@ public void Generate(string diagramPath, string outputDir)
 
         string path = Path.Combine(outputDir, $"{smName}.sim.html");
 
+        string diagramEventNamesArray = OrganizeEventNamesIntoJsArray(diagramEventNames);
+
+        var sb = new StringBuilder();
+        HtmlRenderer.Render(sb,
+            smName: smName,
+            mocksCode: mocksWriter.ToString(),
+            mermaidCode: mermaidCodeWriter.ToString(),
+            jsCode: fileCapturer.CapturedCode,
+            diagramEventNamesArray: diagramEventNamesArray);
+        codeFileWriter.WriteFile(path, code: sb.ToString());
+    }
+
+    private static string OrganizeEventNamesIntoJsArray(HashSet<string> unOrderedEventNames)
+    {
+        string? doEvent = null;
+        List<string> eventNames = new();
+
+        foreach (var name in unOrderedEventNames)
+        {
+            if (TriggerHelper.IsDoEvent(name))
+            {
+                doEvent = name;
+            }
+            else
+            {
+                eventNames.Add(name);
+            }
+        }
+
+        // sort non-do events
+        eventNames.Sort(StringComparer.OrdinalIgnoreCase);  // case-insensitive sort
```

Comment:
I'm not super up on c# syntax, but what about just implementing your own comparator? something like:
```
if(TriggerHelper.IsDoEvent(a)) {
return 0;
} else if TriggerHelper.IsDoEvent(b)) {
return 1;
} else {
return StringComparer.OrdinalIgnoreCase(a,b)
}
```

Reply:
> I'm not super up on c# syntax

Me neither :) It would take me a bit of googling to make it work. I think it would roughly end up being about the same for performance and amount of code (create class, instantiate instance of StringComparer.OrdinalIgnoreCase, checks...).

The comparator approach might end up being a bit slower as it calls `TriggerHelper.IsDoEvent()` check twice as much and that method is currently not implemented very efficiently (trims string, converts to lowercase, compares).

A good idea to keep in mind for future though.